core/src/test/java/org/infinispan/tx/exception/ReplicationTxExceptionTest.java

```diff
@@ -34,15 +35,18 @@ protected void createCacheManagers() throws Throwable {
    public void testReplicationFailure() throws Exception {
       Cache<?, ?> cache = cache(0);
       ControlledRpcManager controlledRpcManager = ControlledRpcManager.replaceRpcManager(cache);
-      Future<Void> future = fork(() -> controlledRpcManager.expectCommand(VersionedPrepareCommand.class).fail());
+      Future<Void> future = fork(() -> {
+         controlledRpcManager.expectCommand(VersionedPrepareCommand.class).fail();
+         controlledRpcManager.expectCommand(RollbackCommand.class).send().receiveAll();
+      });
       try {
          TransactionManager tm = cache(0).getAdvancedCache().getTransactionManager();
          tm.begin();
          cache(0).put("k0", "v");
          Exceptions.expectException(RollbackException.class, tm::commit);
       } finally {
-         controlledRpcManager.revertRpcManager(cache);
          future.get(30, TimeUnit.SECONDS);
+         controlledRpcManager.revertRpcManager();
```

Comment:
I think it would be better to put the `future.get()` inside the `try` block, so `revertRpcManager()` isn't skipped if it times out.

WDYT about moving the tx in the `fork()` and the `expectCommand()+future.get()` in the `try` body, seems all the other tests do it that way?

Reply:
OK, though `fork()` itself shouldn't ever fail...

Reply:
True, I was only thinking it would look better if starting the future and waiting for it were in the same block. I won't complain if you don't feel the same way :)